third_party/mbedtls/repo/CMakeLists.txt

```diff
@@ -211,6 +211,7 @@ endif(CMAKE_COMPILER_IS_GNU)
 
 if(CMAKE_COMPILER_IS_CLANG)
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Wwrite-strings -Wpointer-arith -Wimplicit-fallthrough -Wshadow -Wvla -Wformat=2 -Wno-format-nonliteral")
+    Set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-but-set-variable -Wno-unknown-warning-option")
```

Comment:
Is it possible to modify [third_party/mbedtls/CMakeLists.txt](https://github.com/openthread/openthread/blob/main/third_party/mbedtls/CMakeLists.txt) instead? We should try to keep the mbedtls source unmodified.

Reply:
I wanted to do that first, but I could not get it to work.

I tried setting `CMAKE_C_FLAGS` to `-Wno-unused-but-set-variable -Wno-unknown-warning-option` in  `third_party/mbedtls/CMakeLists.txt` but then the problem I faced is that the code here does:

```set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra ...```

which appends the `-Wall` after the previous value and `-Wno-unused-but-set-variable` option and it seems that order matter and `-Wall` cancel all previous `-Wnoxxx` options.

Not sure if there is a way to make it work.

Reply:
@jwhui , I changed the PR to use `MBEDTLS_FATAL_WARNINGS` option instead (so that compiler warnings are not treated as errors).

We can set it in the [third_party/mbedtls/CMakeLists.txt](https://github.com/openthread/openthread/blob/main/third_party/mbedtls/CMakeLists.txt) (and not need to change any files under the `repo` folder). The downside is warnings are not treated as error when building `mbedtls` library (though we still see them as warning) but I think this is a better approach.

Reply:
SGTM. Thanks!